### PR TITLE
fix: Update ECS deployment guide to use Cloudformation

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

This updated deployment guide makes it easy to deploy the ECS compute and scheduling functionality while also supporting injecting the necessary env variable for use with AWS Marketplace billing

### CODE REVIEW

File: website/pages/docs/deployment/ecs.md [hunk 1]
```diff
@@ -51,24 +51,8 @@ In order to inject the config file into the prebuilt container you will have to
...
-## Step 3: Create a Log Group
-The next step is to create a log group for your ECS task. To create a log group, use the following command:
-
-Prior to running replace `<REPLACE_WITH_LOG_GROUP_NAME>` with the name you want to give to your log group.
-```bash
-aws logs create-log-group --log-group-name <REPLACE_WITH_LOG_GROUP_NAME>
-```
-
-## Step 4: Create Secret to CloudQuery API Key
+## Step 2: Create Secret to CloudQuery API Key
```
Comment (erezrokah) [inline website/pages/docs/deployment/ecs.md:55]:
```suggestion
## Step 2: Create a Secret for the CloudQuery API Key
```

File: website/pages/docs/deployment/ecs.md [hunk 2]
```diff
@@ -78,189 +62,227 @@ Once you have a CloudQuery API Key you are going to create a Secret in AWS Secre
...
-}
-```
-
-Use the file you just created to create an IAM role for your ECS task. To create an IAM role, use the following command:
-
-```bash
-# Prior to running the following command, make sure you have replaced the <REPLACE_WITH_TASK_ROLE_NAME> placeholder with the name of the IAM role you want to create.
-aws iam create-role --role-name <REPLACE_WITH_TASK_ROLE_NAME> --assume-role-policy-document file://task-role-trust-policy.json;
-```
+## Step 3: Create CloudFormation template
```
Comment (erezrokah) [inline website/pages/docs/deployment/ecs.md:70]:
```suggestion
## Step 3: Create a CloudFormation template
```

File: website/pages/docs/deployment/ecs.md [hunk 3]
```diff
@@ -78,189 +62,227 @@ Once you have a CloudQuery API Key you are going to create a Secret in AWS Secre
...
-  - `<REPLACE_WITH_CQ_BASE64_ENCODED_CONFIG>` : The base64 encoded version of the CloudQuery configuration file you created in Step 1.
-  - `<REPLACE_WITH_LOG_GROUP_NAME>` : The name of the CloudWatch log group you created in Step 4.
-  - `<REPLACE_WITH_AWS_REGION>` : The AWS region where you created the CloudWatch log group in Step 4.
-  - `<REPLACE_WITH_PREFIX_FOR_STREAM>` : The prefix you want to use for the CloudWatch log stream.
-  - `<REPLACE_WITH_TASK_FAMILY_NAME>` : The name of the task family you want to use.
 
 
-Once you have modified the `task-definition.json` file to include the correct values for your environment, you can register the task definition with AWS ECS using the following command:
-```bash
+## Step 4: Run a test CloudQuery sync
```
Comment (erezrokah) [inline website/pages/docs/deployment/ecs.md:277]:
```suggestion
## Step 4: Run a CloudQuery sync
```

**Review by erezrokah — COMMENTED** [review]
Looks good added a few comments

File: website/pages/docs/deployment/ecs.md [hunk 4]
```diff
@@ -78,189 +62,227 @@ Once you have a CloudQuery API Key you are going to create a Secret in AWS Secre
...
+    Value: !Ref ECSCluster
+  TaskArn:
+    Value: !Ref ScheduledWorkerTask
 ```
 
-Replace the `REPLACE_WITH_S3_DESTINATION_BUCKET` placeholder with the name of the S3 bucket you want to use to store the data. 
-
 
-This policy will allow the ECS task to write data to the S3 bucket you specified while also ensuring that CloudQuery never has access to any of your data.
+## Step 3: Deploy the CloudFormation Template
```
Comment (marianogappa) [inline website/pages/docs/deployment/ecs.md:267]: Step 3 is duplicated here

File: website/pages/docs/deployment/ecs.md (hunk 1 above)
Comment (marianogappa) [reply website/pages/docs/deployment/ecs.md:55]: Btw in the previous section (sorry couldn't add a comment) there's no Windows version for the base64 encode. Apparently it's:

```
echo "your_string" | certutil -encodehex -f - 0
```

File: website/pages/docs/deployment/ecs.md [hunk 5]
```diff
@@ -78,189 +62,227 @@ Once you have a CloudQuery API Key you are going to create a Secret in AWS Secre
...
+  # Scheduler Configurations
+  Schedule:
+    Type: AWS::Scheduler::Schedule
+    Properties: 
+      FlexibleTimeWindow: 
+        Mode: "OFF"
+      ScheduleExpression: !Ref ScheduleExpression
+      State: ENABLED
+      Target: 
+          Arn: !GetAtt ECSCluster.Arn
```
Comment (marianogappa) [inline website/pages/docs/deployment/ecs.md:160]: I think this is a problem here with the inconsistent spacing. It seems to be 4 spaces here and 2 spaces mostly elsewhere.

File: website/pages/docs/deployment/ecs.md [hunk 6]
```diff
@@ -78,189 +62,227 @@ Once you have a CloudQuery API Key you are going to create a Secret in AWS Secre
...
+        - FARGATE
+      NetworkMode: awsvpc
+      Cpu: 1024
+      Memory: 2GB
+      ExecutionRoleArn: !GetAtt ExecutionRole.Arn
+      TaskRoleArn: !GetAtt ExecutionRole.Arn
+      ContainerDefinitions:
+        - Essential: 'true'
+          Command: 
+            - "echo $CQ_CONFIG| base64 -d  > ./file.yml;/app/cloudquery sync ./file.yml --log-console --log-format json"
```
Comment (marianogappa) [inline website/pages/docs/deployment/ecs.md:123]: I think `base64 -d` is Linux only (`-D` on Mac I think); is this cloudformation template only applied on Linux instances?

File: website/pages/docs/deployment/ecs.md (hunk 1 above)
Comment (bbernays) [reply website/pages/docs/deployment/ecs.md:55]: This tutorial has been up for over a year and people have not asked about that. If we want to add windows support I think that can be done in a different PR

File: website/pages/docs/deployment/ecs.md (hunk 6 above)
Comment (bbernays) [reply website/pages/docs/deployment/ecs.md:123]: This is only Linux

File: website/pages/docs/deployment/ecs.md (hunk 5 above)
Comment (bbernays) [reply website/pages/docs/deployment/ecs.md]: fixed in [128340e](https://github.com/cloudquery/cloudquery/pull/18964/commits/128340e7b2333502de16087e63b21d154ed4c2bc)

File: website/pages/docs/deployment/ecs.md (hunk 4 above)
Comment (bbernays) [reply website/pages/docs/deployment/ecs.md]: Fixed in [caeb720](https://github.com/cloudquery/cloudquery/pull/18964/commits/caeb720af516f894e04dd7b5746a7ded81b81bed)

**Review by marianogappa — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Pull request merged**